### PR TITLE
Don't show scene group progress for 1 scene

### PR DESCRIPTION
**Pull request by KoBeWi**
The new progress bar for updating scene groups (#89739) appears too often. It shows when you open project and every time you save a scene (in which case 2 progress bards appear).
image
This PR makes it appear only when updating more than 1 scene, so it's less annoying.

### CODE REVIEW

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -1664,7 +1664,10 @@ void EditorFileSystem::_queue_update_script_class(const String &p_path) {
 }
 
 void EditorFileSystem::_update_scene_groups() {
-	EditorProgress ep("update_scene_groups", TTR("Update Scene Groups"), update_scene_paths.size());
+	EditorProgress *ep = nullptr;
+	if (update_scene_paths.size() > 1) {
+		ep = memnew(EditorProgress("update_scene_groups", TTR("Update Scene Groups"), update_scene_paths.size()));
+	}
 	int step_count = 0;
```
Comment (akien-mga) [inline editor/editor_file_system.cpp:1671]: Maybe this should go in the `if` too since it's not used otherwise?

**Review by akien-mga — APPROVED** [review]

File: editor/editor_file_system.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/editor_file_system.cpp:1671]: Variable declaration can't be ifed. (which is also why I had to use pointer to EditorProgress)
To completely avoid it I'd have to repeat the whole code twice in different branches.

File: editor/editor_file_system.cpp [hunk 2]
```diff
@@ -1684,9 +1687,14 @@ void EditorFileSystem::_update_scene_groups() {
...
 
-		ep.step(TTR("Updating Scene Groups..."), step_count++);
+		if (ep) {
+			ep->step(TTR("Updating Scene Groups..."), step_count++);
+		}
 	}
 
+	if (ep) {
+		memdelete(ep);
+	}
```
Comment (AThousandShips) [inline editor/editor_file_system.cpp:1697]: For convenience you can use `memdelete_notnull` here instead

Comment (KoBeWi) [reply editor/editor_file_system.cpp:1697]: Looks like it will be the first usage of this macro, unless I'm missing something 🤔

Comment (AThousandShips) [reply editor/editor_file_system.cpp]: Weird!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**